nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/PeerSelector.java

```diff
@@ -106,6 +106,16 @@ private void restoreInitialPeerStatusCache() {
                     final SiteToSiteTransportProtocol currentProtocol = peerStatusProvider.getTransportProtocol();
                     final SiteToSiteTransportProtocol cachedProtocol = restoredPeerStatusCache.getTransportProtocol();
 
+                    final String currentRemoteInstanceUris = peerStatusProvider.getRemoteInstanceUris();
+                    final String cachedRemoteInstanceUris = restoredPeerStatusCache.getRemoteInstanceUris();
+
+                    // If the remote instance URIs have changed, clear the cache
+                    if (!currentRemoteInstanceUris.equals(cachedRemoteInstanceUris)) {
+                        logger.warn("Discard stored peer statuses in {} because remote instance URIs has changed from {} to {}",
```

Comment:
No reason to warn here. This is a normal and expected thing. Can use INFO or really even DEBUG.

Reply:
changed to INFO